store/copr/coprocessor.go

```diff
@@ -753,8 +788,12 @@ func (worker *copIteratorWorker) handleTaskOnce(bo *Backoffer, task *copTask, ch
 		return worker.handleCopStreamResult(bo, rpcCtx, resp.Resp.(*tikvrpc.CopStreamResponse), task, ch, costTime)
 	}
 
+	var lastRange *coprocessor.KeyRange
+	if task.paging {
+		lastRange = resp.Resp.(*coprocessor.Response).Range
+	}
 	// Handles the response for non-streaming copTask.
-	return worker.handleCopResponse(bo, rpcCtx, &copResponse{pbResp: resp.Resp.(*coprocessor.Response)}, cacheKey, cacheValue, task, ch, nil, costTime)
+	return worker.handleCopResponse(bo, rpcCtx, &copResponse{pbResp: resp.Resp.(*coprocessor.Response)}, cacheKey, cacheValue, task, ch, lastRange, costTime)
```

Comment:
The `lastRange` parameter is totally useless after we fix copr streaming. I prefer to add a `handleCopPagingResult()` to separate the paging logic from `handleCopResponse()`.

Reply:
Seems calculating the remaining ranges when meeting a lock still requires the `lastRange` parameter for both streaming and paging modes. I'll invoke `handleCopResponse` from `handleCopPagingResult`, it'll be more clear.

Reply:
> Seems calculating the remaining ranges when meeting a lock still requires the lastRange parameter for streaming.

Oh, yes. `lastRange` is useful for streaming requests, but it's useless for paging requests. `resp.Resp.(*coprocessor.Response).Range` is nil when meets a lock. Paging and unary requests can retry the request directly. without invoking `buildCopTasksFromRemain()`.